include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
+        std::string str = std::to_string(t);
+        str.erase(str.find_last_not_of('0') + 1, std::string::npos);
+        if (str.back() == '.') {
```

Comment:
> iOS system region currently has no effect on what decimal separator is used here.

Yes, this is more likely to be a problem on desktop systems, that sort of thing. There’s also potentially a distinction between the current C locale and current C++ locale, and the C locale is more likely to be affected.

Rather than delve into the complexities of number formatting, I think it’d be more robust to rely on existing string formatters. I’m sure we can find something between C-style `sprintf()` and C++-style [`stringstream`](http://www.cplusplus.com/reference/sstream/stringstream/).